File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/mix/MixOrderApi.java
package com.bitget.openapi.api.mix;


import com.bitget.openapi.dto.request.mix.MixCancelBatchOrdersReq;
import com.bitget.openapi.dto.request.mix.MixCancelOrderReq;
import com.bitget.openapi.dto.request.mix.MixPlaceOrderReq;
import com.bitget.openapi.dto.request.mix.PlaceBatchOrderReq;
import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: Contract order api
 */
public interface MixOrderApi {

    /**
     * place an order
     * @param mixPlaceOrderReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/order/placeOrder")
    Call<ResponseResult> placeOrder(@Body MixPlaceOrderReq mixPlaceOrderReq);

    /**
     * place an order
     * @param mixPlaceOrderReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/order/proportionOrder")
    Call<ResponseResult> proportionOrder(@Body MixPlaceOrderReq mixPlaceOrderReq);

    /**
     * Place orders in batches
     * @param placeBatchOrderReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/order/batch-orders")
    Call<ResponseResult> batchOrders(@Body PlaceBatchOrderReq placeBatchOrderReq);

    /**
     * cancel the order
     * @param mixCancelOrderReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/order/cancel-order")
    Call<ResponseResult> cancelOrder(@Body MixCancelOrderReq mixCancelOrderReq);

    /**
     * Batch cancellation
     * @param mixCancelBatchOrdersReq
     * @return ResponseResult
     */
    @POST("/api/mix/v1/order/cancel-batch-orders")
    Call<ResponseResult> cancelBatchOrder(@Body MixCancelBatchOrdersReq mixCancelBatchOrdersReq);

    /**
     * Get Historical Delegation
     * @param symbol
     * @param startTime
     * @param endTime
     * @param pageSize
     * @param lastEndId
     * @param isPre
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/history")
    Call<ResponseResult> history(@Query("symbol")String symbol,
                                             @Query("startTime")String startTime,
                                             @Query("endTime")String endTime,
                                             @Query("pageSize") int pageSize,
                                             @Query("lastEndId")String lastEndId,
                                             @Query("isPre") boolean isPre);

    /**
     * Get all Historical Delegation
     * @param productType
     * @param startTime
     * @param endTime
     * @param pageSize
     * @param lastEndId
     * @param isPre
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/historyProductType")
    Call<ResponseResult> historyProductType(@Query("productType")String productType,
                                 @Query("startTime")String startTime,
                                 @Query("endTime")String endTime,
                                 @Query("pageSize") int pageSize,
                                 @Query("lastEndId")String lastEndId,
                                 @Query("isPre") boolean isPre);

    /**
     * Get the current delegate
     * @param symbol
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/current")
    Call<ResponseResult> current(@Query("symbol")String symbol);

    /**
     * Get the all current delegate
     * @param symbol
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/marginCoinCurrent")
    Call<ResponseResult> marginCoinCurrent(@Query("productType")String productType,
                                           @Query("symbol")String symbol);

    /**
     * Get order details
     * @param symbol
     * @param orderId
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/detail")
    Call<ResponseResult> detail(@Query("symbol")String symbol,
                                @Query("orderId")String orderId);

    /**
     * Query transaction details
     * @param symbol
     * @param orderId
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/fills")
    Call<ResponseResult> fills(@Query("symbol")String symbol,
                               @Query("orderId")String orderId);

    /**
     * Query productType transaction details
     * @return ResponseResult
     */
    @GET("/api/mix/v1/order/allFills")
    Call<ResponseResult> allFills(@Query("productType")String productType,
                               @Query("startTime")String startTime,
                                  @Query("endTime")String endTime,
                                  @Query("lastEndId")String lastEndId);
}

File: bitget-php-sdk-api/src/api/mix/MixOrderApi.php
<?php


namespace bitget\api\mix;


use bitget\internal\BitgetApiClient;
use bitget\model\mix\order\BatchOrdersReq;
use bitget\model\mix\order\CancelBatchOrderReq;
use bitget\model\mix\order\CancelOrderReq;
use bitget\model\mix\order\PlaceOrderReq;

class MixOrderApi
{
    const BASE_URL = "/api/mix/v1/order";

    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }
    /**
     * place an order
     * @param $placeOrderReq
     * @return string
     */
    public function placeOrder(PlaceOrderReq $placeOrderReq): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/placeOrder", $placeOrderReq);
    }
    /**
     * Place orders in batches
     * @param $batchOrdersReq
     * @return string
     */
    public function batchOrders(BatchOrdersReq $batchOrdersReq): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/batch-orders", $batchOrdersReq);
    }
    /**
     * cancel the order
     * @param $cancelOrderReq
     * @return string
     */
    public function cancelOrder(CancelOrderReq $cancelOrderReq): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/cancel-order", $cancelOrderReq);
    }
    /**
     * Batch cancellation
     * @param $cancelBatchOrderReq
     * @return string
     */
    public function cancelBatchOrder(CancelBatchOrderReq $cancelBatchOrderReq): string
    {
        return $this->BitgetApiClient->doPost(self::BASE_URL . "/cancel-batch-orders", $cancelBatchOrderReq);
    }
    /**
     * Get Historical Delegation
     * @param $symbol
     * @param $startTime
     * @param $endTime
     * @param $pageSize
     * @param $lastEndId
     * @param $isPre
     * @return string
     */
    public function history(string $symbol, string $startTime, string $endTime, string $pageSize, string $lastEndId, string $isPre): string
    {
        $params = array("symbol" => $symbol, "startTime" => $startTime, "endTime" => $endTime, "pageSize" => $pageSize, "lastEndId" => $lastEndId, "isPre" => $isPre);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/history", $params);
    }

    /**
     * Get the current delegate
     * @param $symbol
     * @return string
     */
    public function current(string $symbol): string
    {
        $params = array("symbol" => $symbol);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/current", $params);
    }
    /**
     * Get order details
     * @param $symbol
     * @param $orderId
     * @return string
     */
    public function detail(string $symbol, string $orderId): string
    {
        $params = array("symbol" => $symbol, "orderId" => $orderId);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/detail", $params);
    }
    /**
     * Query transaction details
     * @param $symbol
     * @param $orderId
     * @return string
     */
    public function fills(string $symbol, string $orderId): string
    {
        $params = array("symbol" => $symbol, "orderId" => $orderId);
        return $this->BitgetApiClient->doGet(self::BASE_URL . "/fills", $params);
    }


}